zeroconf: Type '��W.' must end with '.local.'

### The problem

this is happening off and on lately (at startup), not sure when it started, but I believe it was during the beta of current cycle



### What version of Home Assistant Core has the issue?

2023.1.5

### What was the last working version of Home Assistant Core?

2022.12

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Zeroconf

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/zeroconf/

### Diagnostics information

_No response_

### Example YAML snippet


not using `default_config:` but have `zeroconf:` listed in configuration.yaml



### Anything in the logs that might be useful for us?

```txt
Logger: homeassistant
Source: components/zeroconf/__init__.py:401 
First occurred: 12:09:32 (2 occurrences) 
Last logged: 12:19:22

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/zeroconf/__init__.py", line 401, in _process_service_update
    async_service_info = AsyncServiceInfo(service_type, name)
  File "/usr/local/lib/python3.10/site-packages/zeroconf/_services/info.py", line 141, in __init__
    if not type_.endswith(service_type_name(name, strict=False)):
  File "/usr/local/lib/python3.10/site-packages/zeroconf/_utils/name.py", line 97, in service_type_name
    raise BadTypeInNameException(f"Type '{type_}' must end with '{_LOCAL_TRAILER}'")
zeroconf._exceptions.BadTypeInNameException: Type '��W.' must end with '.local.'
```


### Additional information

_No response_

```
E       AssertionError: assert 5 == 0
E        +  where 5 = len([record[ptr,in,_hap._tcp.local.]=4500/4499,��W., record[txt,in-unique,��W.]=4500/4499,b'\x07c#=200'..., record[srv,in-unique,��W.]=120/119,TRADFRI-Gateway-dcefcaba509f.local.:80, record[a,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=120/119,192.168.1.99, record[quada,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=0/0,f0:3eb0:7a80:12ea:7a80:12ea:7a80:12ea])
E        +    where [record[ptr,in,_hap._tcp.local.]=4500/4499,��W., record[txt,in-unique,��W.]=4500/4499,b'\x07c#=200'..., record[srv,in-unique,��W.]=120/119,TRADFRI-Gateway-dcefcaba509f.local.:80, record[a,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=120/119,192.168.1.99, record[quada,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=0/0,f0:3eb0:7a80:12ea:7a80:12ea:7a80:12ea] = <DNSIncoming:{id=0, flags=33792, truncated=False, n_q=0, n_ans=5, n_auth=0, n_add=0, questions=[], answers=[record[ptr,in,_hap._tcp.local.]=4500/4499,��W., record[txt,in-unique,��W.]=4500/4499,b'\x07c#=200'..., record[srv,in-unique,��W.]=120/119,TRADFRI-Gateway-dcefcaba509f.local.:80, record[a,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=120/119,192.168.1.99, record[quada,in-unique,TRADFRI-Gateway-dcefcaba509f.local.]=0/0,f0:3eb0:7a80:12ea:7a80:12ea:7a80:12ea]}>.answers

```

So this is technically a valid name so we can't filter it out at the `zeroconf` level but obvious its garbage.

--- a/homeassistant/components/homekit_controller/manifest.json
+++ b/homeassistant/components/homekit_controller/manifest.json
@@ -3,7 +3,7 @@
   "name": "HomeKit Controller",
   "config_flow": true,
   "documentation": "https://www.home-assistant.io/integrations/homekit_controller",
-  "requirements": ["aiohomekit==2.4.5"],
+  "requirements": ["aiohomekit==2.4.6"],
   "zeroconf": ["_hap._tcp.local.", "_hap._udp.local."],
   "bluetooth": [{ "manufacturer_id": 76, "manufacturer_data_start": [6] }],
   "dependencies": ["bluetooth_adapters", "zeroconf"],

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -174,7 +174,7 @@ aioguardian==2022.07.0
 aioharmony==0.2.9
 
 # homeassistant.components.homekit_controller
-aiohomekit==2.4.5
+aiohomekit==2.4.6
 
 # homeassistant.components.emulated_hue
 # homeassistant.components.http

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -158,7 +158,7 @@ aioguardian==2022.07.0
 aioharmony==0.2.9
 
 # homeassistant.components.homekit_controller
-aiohomekit==2.4.5
+aiohomekit==2.4.6
 
 # homeassistant.components.emulated_hue
 # homeassistant.components.http

